src/Databases/MySQL/DatabaseMySQL.cpp

```diff
@@ -144,6 +144,7 @@ ASTPtr DatabaseMySQL::getCreateTableQueryImpl(const String & table_name, Context
     auto table_storage_define = database_engine_define->clone();
     {
         ASTStorage * ast_storage = table_storage_define->as<ASTStorage>();
+        ast_storage->engine->kind = ASTFunction::Kind::TABLE_ENGINE;
```

Comment:
Do you know why this doesn't this come from the clone and needs to be set explicitly?

Reply:
Because the original `database_engine_define` is a database engine, so the type is `ASTFunction::Kind::DATABASE_ENGINE`.